### ISSUE TITLE
crash when pulling with flatpak

### ISSUE THREAD

**Issue by fcrozat** [ostreedev/ostree#370]
I got a crash when pulling from Flatpak, using ostree 2016.5 on openSUSE Leap 42.1:

```
Program terminated with signal SIGSEGV, Segmentation fault.
#0  0x00007f37a1d81cf5 in g_mutex_lock (mutex=mutex@entry=0x6133636366366161) at gthread-posix.c:1336
1336	  if G_UNLIKELY (g_atomic_int_add (&mutex->i[0], 1) != 0)
Missing separate debuginfos, use: zypper install glib-networking-debuginfo-2.44.0-2.4.x86_64 glibc-locale-debuginfo-2.19-19.1.x86_64 gnome-keyring-debuginfo-3.16.0-7.1.x86_64 gvfs-debuginfo-1.24.3-7.2.x86_64 libXau6-debuginfo-1.0.8-6.1.x86_64 libarchive13-debuginfo-3.1.2-10.1.x86_64 libassuan0-debuginfo-2.1.1-3.6.x86_64 libattr1-debuginfo-2.4.47-5.4.x86_64 libblkid1-debuginfo-2.25-15.2.x86_64 libbz2-1-debuginfo-1.0.6-32.1.x86_64 libgpgme11-debuginfo-1.6.0-9.1.x86_64 libgsystem0-debuginfo-2015.2-4.2.x86_64 liblzma5-debuginfo-5.0.5-3.5.x86_64 libmount1-debuginfo-2.25-15.2.x86_64 libopenssl1_0_0-debuginfo-1.0.1i-15.1.x86_64 libseccomp2-debuginfo-2.1.1-5.1.x86_64 libselinux1-debuginfo-2.3-5.1.x86_64 libsoup-2_4-1-debuginfo-2.50.0-5.1.x86_64 libsqlite3-0-debuginfo-3.8.10.2-5.2.x86_64 libudev1-debuginfo-210-95.1.x86_64 libuuid1-debuginfo-2.25-15.2.x86_64 libxml2-2-debuginfo-2.9.1-19.1.x86_64 systemd-debuginfo-210-95.1.x86_64
(gdb) bt
#0  0x00007f37a1d81cf5 in g_mutex_lock (mutex=mutex@entry=0x6133636366366161) at gthread-posix.c:1336
#1  0x00007f37a1d3d458 in g_source_destroy_internal (source=0x16d12c0, context=0x6133636366366161, have_lock=0) at gmain.c:1159
#2  0x00007f37a26474c8 in  () at /usr/lib64/libsoup-2.4.so.1
#3  0x00007f37a20187f3 in g_object_unref (_object=0x12d3240) at gobject.c:3137
#4  0x00007f37a28f3e46 in thread_closure_unref (thread_closure=0x12c35e0) at src/libostree/ostree-fetcher.c:137
#5  0x00007f37a28f402f in ostree_fetcher_session_thread (data=0x12c35e0) at src/libostree/ostree-fetcher.c:467
#6  0x00007f37a1d64f65 in g_thread_proxy (data=0x12c1450) at gthread.c:764
#7  0x00007f37a16b60a4 in start_thread (arg=0x7f379aca4700) at pthread_create.c:309
#8  0x00007f37a13eafed in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:111


(gdb) thread apply all bt

Thread 4 (Thread 0x7f379d3c2700 (LWP 20380)):
#0  0x00007f37a13e2bbd in poll () at ../sysdeps/unix/syscall-template.S:81
#1  0x00007f37a1d3fe64 in g_main_context_iterate (priority=2147483647, n_fds=2, fds=0x12f15b0, timeout=-1, context=0x148fff0) at gmain.c:4103
#2  0x00007f37a1d3fe64 in g_main_context_iterate (context=0x148fff0, block=block@entry=1, dispatch=dispatch@entry=1, self=<optimized out>) at gmain.c:3803
#3  0x00007f37a1d4019a in g_main_loop_run (loop=0x12e4440) at gmain.c:4002
#4  0x00007f37a2335426 in gdbus_shared_thread_func (user_data=0x148ffc0) at gdbusprivate.c:274
#5  0x00007f37a1d64f65 in g_thread_proxy (data=0x12c11e0) at gthread.c:764
#6  0x00007f37a16b60a4 in start_thread (arg=0x7f379d3c2700) at pthread_create.c:309
#7  0x00007f37a13eafed in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:111

Thread 3 (Thread 0x7f379b6dd700 (LWP 20365)):
#0  0x00007f37a13e6f79 in syscall () at ../sysdeps/unix/sysv/linux/x86_64/syscall.S:38
#1  0x00007f37a1d81f3e in g_cond_wait_until (cond=cond@entry=0x12d27a8, mutex=mutex@entry=0x12d27a0, end_time=end_time@entry=29989651355) at gthread-posix.c:1442
#2  0x00007f37a1d15dc1 in g_async_queue_pop_intern_unlocked (queue=queue@entry=0x12d27a0, wait=wait@entry=1, end_time=end_time@entry=29989651355) at gasyncqueue.c:422
#3  0x00007f37a1d1634b in g_async_queue_timeout_pop (queue=0x12d27a0, timeout=timeout@entry=15000000) at gasyncqueue.c:543
#4  0x00007f37a1d659ac in g_thread_pool_thread_proxy () at gthreadpool.c:167
#5  0x00007f37a1d659ac in g_thread_pool_thread_proxy (data=<optimized out>) at gthreadpool.c:364
#6  0x00007f37a1d64f65 in g_thread_proxy (data=0x12c1770) at gthread.c:764
#7  0x00007f37a16b60a4 in start_thread (arg=0x7f379b6dd700) at pthread_create.c:309
#8  0x00007f37a13eafed in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:111

Thread 2 (Thread 0x7f37a2f13880 (LWP 20362)):
#0  0x00007f37a137e0d5 in malloc_consolidate (av=av@entry=0x7f37a16a8620 <main_arena>) at malloc.c:4142
#1  0x00007f37a137f4e8 in _int_malloc (av=av@entry=0x7f37a16a8620 <main_arena>, bytes=bytes@entry=4128) at malloc.c:3422
#2  0x00007f37a13807bb in _int_memalign (av=av@entry=0x7f37a16a8620 <main_arena>, alignment=alignment@entry=2048, bytes=bytes@entry=2033) at malloc.c:4402
#3  0x00007f37a1380a46 in memalign_check (alignment=2048, bytes=2032, caller=<optimized out>) at hooks.c:426
#4  0x00007f37a1383612 in __posix_memalign (memptr=memptr@entry=0x7ffda99d7748, alignment=alignment@entry=2048, size=size@entry=2032) at malloc.c:5015
#5  0x00007f37a1d11c13 in slab_allocator_alloc_chunk (memsize=2032, alignment=2048) at gslice.c:1378
#6  0x00007f37a1d11c13 in slab_allocator_alloc_chunk (allocator=0x7f37a20022c0 <allocator>, chunk_size=144, ix=8) at gslice.c:1252
#7  0x00007f37a1d11c13 in slab_allocator_alloc_chunk (chunk_size=chunk_size@entry=144) at gslice.c:1297
#8  0x00007f37a1d5b647 in g_slice_alloc (countp=0x12a6d78, ix=8) at gslice.c:731
#9  0x00007f37a1d5b647 in g_slice_alloc (tmem=<optimized out>, ix=8) at gslice.c:801
#10 0x00007f37a1d5b647 in g_slice_alloc (mem_size=mem_size@entry=136) at gslice.c:996
#11 0x00007f37a1d5b6ae in g_slice_alloc0 (mem_size=mem_size@entry=136) at gslice.c:1032
#12 0x00007f37a28f47e2 in _ostree_fetcher_constructed (object=0x1609200 [OstreeFetcher]) at src/libostree/ostree-fetcher.c:533
#13 0x00007f37a2019092 in g_object_new_internal (class=class@entry=0x12ca500, params=params@entry=0x7ffda99d79c0, n_params=1) at gobject.c:1814
#14 0x00007f37a201af24 in g_object_new_valist (object_type=object_type@entry=19702288, first_property_name=first_property_name@entry=0x7f37a290a288 "config-flags", var_args=var_args@entry=0x7ffda99d7b18)
    at gobject.c:2033
#15 0x00007f37a201b304 in g_object_new (object_type=19702288, first_property_name=first_property_name@entry=0x7f37a290a288 "config-flags") at gobject.c:1617
#16 0x00007f37a28f46b4 in _ostree_fetcher_new (tmpdir_dfd=24, flags=OSTREE_FETCHER_FLAGS_NONE) at src/libostree/ostree-fetcher.c:607
#17 0x00007f37a28c95cd in _ostree_repo_remote_new_fetcher (self=self@entry=0x12c4320 [OstreeRepo], remote_name=remote_name@entry=0x7ffda99d9a61 "gnome", error=error@entry=0x7ffda99d8330)
    at src/libostree/ostree-repo.c:518
#18 0x00007f37a28f9ad5 in ostree_repo_pull_with_options (self=self@entry=0x12c4320 [OstreeRepo], remote_name_or_baseurl=remote_name_or_baseurl@entry=0x7ffda99d9a61 "gnome", options=<optimized out>, progress=progress@entry=0x12c1630 [OstreeAsyncProgress], cancellable=cancellable@entry=0x0, error=error@entry=0x7ffda99d8330) at src/libostree/ostree-repo-pull.c:2372
#19 0x00007f37a28fb781 in ostree_repo_pull_one_dir (self=self@entry=0x12c4320 [OstreeRepo], remote_name=remote_name@entry=0x7ffda99d9a61 "gnome", dir_to_pull=dir_to_pull@entry=0x0, refs_to_fetch=refs_to_fetch@entry=0x7ffda99d8190, flags=flags@entry=OSTREE_REPO_PULL_FLAGS_MIRROR, progress=progress@entry=0x12c1630 [OstreeAsyncProgress], cancellable=cancellable@entry=0x0, error=error@entry=0x7ffda99d8330)
    at src/libostree/ostree-repo-pull.c:1773
#20 0x00007f37a28fb7c2 in ostree_repo_pull (self=self@entry=0x12c4320 [OstreeRepo], remote_name=remote_name@entry=0x7ffda99d9a61 "gnome", refs_to_fetch=refs_to_fetch@entry=0x7ffda99d8190, flags=flags@entry=OSTREE_REPO_PULL_FLAGS_MIRROR, progress=progress@entry=0x12c1630 [OstreeAsyncProgress], cancellable=cancellable@entry=0x0, error=error@entry=0x7ffda99d8330) at src/libostree/ostree-repo-pull.c:1747
#21 0x000000000042154f in flatpak_dir_pull (self=self@entry=0x12c1000 [FlatpakDir], repository=repository@entry=0x7ffda99d9a61 "gnome", ref=ref@entry=0x1389eb0 "runtime/org.gnome.Platform/x86_64/3.20", subpaths=---Type <return> to continue, or q <return> to quit---
subpaths@entry=0x0, repo=repo@entry=0x12c4320 [OstreeRepo], flags=flags@entry=OSTREE_REPO_PULL_FLAGS_MIRROR, progress=0x12c1630 [OstreeAsyncProgress], 
    progress@entry=0x0, cancellable=cancellable@entry=0x0, error=error@entry=0x7ffda99d8330) at common/flatpak-dir.c:1244
#22 0x0000000000426632 in flatpak_dir_install (self=0x12c1000 [FlatpakDir], no_pull=0, no_deploy=0, ref=ref@entry=0x1389eb0 "runtime/org.gnome.Platform/x86_64/3.20", remote_name=remote_name@entry=0x7ffda99d9a61 "gnome", subpaths=0x0, progress=progress@entry=0x0, cancellable=cancellable@entry=0x0, error=0x7ffda99d8330) at common/flatpak-dir.c:3000
#23 0x000000000041049f in flatpak_builtin_install (argc=4, argv=0x7ffda99d8468, cancellable=0x0, error=<optimized out>) at app/flatpak-builtins-install.c:244
#24 0x000000000040e5cd in flatpak_run (argc=4, argc@entry=5, argv=argv@entry=0x7ffda99d8468, res_error=res_error@entry=0x7ffda99d8360) at app/flatpak-main.c:337
#25 0x000000000040de9c in main (argc=5, argv=0x7ffda99d8468) at app/flatpak-main.c:422

Thread 1 (Thread 0x7f379aca4700 (LWP 20381)):
#0  0x00007f37a1d81cf5 in g_mutex_lock (mutex=mutex@entry=0x6133636366366161) at gthread-posix.c:1336
#1  0x00007f37a1d3d458 in g_source_destroy_internal (source=0x16d12c0, context=0x6133636366366161, have_lock=0) at gmain.c:1159
#2  0x00007f37a26474c8 in  () at /usr/lib64/libsoup-2.4.so.1
#3  0x00007f37a20187f3 in g_object_unref (_object=0x12d3240) at gobject.c:3137
#4  0x00007f37a28f3e46 in thread_closure_unref (thread_closure=0x12c35e0) at src/libostree/ostree-fetcher.c:137
#5  0x00007f37a28f402f in ostree_fetcher_session_thread (data=0x12c35e0) at src/libostree/ostree-fetcher.c:467
#6  0x00007f37a1d64f65 in g_thread_proxy (data=0x12c1450) at gthread.c:764
#7  0x00007f37a16b60a4 in start_thread (arg=0x7f379aca4700) at pthread_create.c:309
#8  0x00007f37a13eafed in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:111
```

**Comment by cgwalters** [comment]
Please try with https://mail.gnome.org/archives/ostree-list/2016-June/msg00006.html

**Comment by fcrozat** [comment]
My mistake, I upgraded ostree on openSUSE to 2016.6 this morning but forgot to install it on my system. I only got the crash one time, with 2016.5 and I couldn't reproduce it. Closing as fix, I'll reopen if I can reproduce it is 2016.6.
